### ISSUE TITLE
Text hides behind keyboard on "Return" press or Paste from clipboard

### ISSUE THREAD

**Issue by PuneetKohli** [hackiftekhar/IQKeyboardManager#498]
Hi,

I'm trying to implement this with a very dynamic view - I have a UITableView which has TextView's of resizable height. 

When I add text to the textview, everything works as expected. But when I press "return" or when I paste text from the clipboard to the textviews, it does not adjust with the keyboard.

I tried a few workarounds, and even made adjustFrame() a public function, calling it whenever there was a return press, or a paste, but it didn't seem to work.

Kindly lead me in the right direction of how to handle this with your library. Cheers!

(Using Swift btw)

**Comment by hackiftekhar** [comment]
Are you using UITableViewController's subclass or you are directly using UITableView? If it's UITableViewController's subclass then I can't do anything because textField hiding issues is already managed by UITableViewController. If you are using UITableView directly then I need to check what is wrong with your code.

**Comment by PuneetKohli** [comment]
I'm using a UITableView. My ViewController is quite complex. It's for a Note taking app that allows both images and text, similar to the Medium app

There is a main view, which contains a tableview (no textfields). There is a tableview (lets refer to this tableview from now onwards) in a hidden panel that comes down on swipe from top of the screen. This tableview contains 2 types of cells
1. TextCell - Which has a resizable textfield 
2. ImageCell - Which are images

Now, whenever I press return/paste into a TextCell, it resizes the TextCell according to the content size, but the keyboard doesn't adjust the view, so everything is still hidden behind the keyboard until I type.

I can think of 2 ways for me to handle this 
1. There would possibly be a way for me to handle specific cases like this and ask your KeyboardManager to recalculate/readjust the views when they occur
2. Have the KeyboardManager take into account enter key or paste situations

Let me know, I can send you a video of this as well.

**Comment by hackiftekhar** [comment]
I understand what issue you are facing. KeyboardManager adjust textField positioned perfectly but when your textField size gets updated then KeyboardManager doesn't update view position. Correct?

I'll try to fix it.

**Comment by PuneetKohli** [comment]
Yes. Correct

**Comment by PuneetKohli** [comment]
Is there any temporary fix while you get to updating the code?
Or atleast if you point me in the right direction I can try figuring it out myself

**Comment by hackiftekhar** [comment]
`-(void)adjustFrame` is the main method responsible for adjusting keyboard position. But it's not a public method. If you didn't used source code then you can give a try to call this method.

**Comment by PuneetKohli** [comment]
Yes I tried implementing that function (made it public) every time I am adjusting the textview height, but it doesn't seem to do anything at the moment.

Using it by calling sharedmanager().adjustFrame()

Could you possibly look into it soon.

**Comment by hackiftekhar** [comment]
Well, I think this is duplicate of #496. Could you please confirm?

**Comment by hackiftekhar** [comment]
I've added `reloadLayoutIfNeeded` method for you, Please check and let me know if it works for you.

**Comment by hackiftekhar** [comment]
One more thing, could you please check if you added `topLayoutConstraint` or `bottomLayoutConstraint` in your view hierarchy which may be affecting the layout?

**Comment by PuneetKohli** [comment]
I am checking the reloadLayoutIfNeeded function now.

As for the constraints - yes it is added.

Also, this is not a duplicate of #496

**Comment by hackiftekhar** [comment]
If you've added topLayoutConstraint or bottomLayoutConstraint then you have to map it with IQLayoutConstraint. because UILayoutConstraint work a bit differently as compared to normal constraint and there is a workaround for that.

**Comment by PuneetKohli** [comment]
The view is completely dynamic, so none of it is via storyboard.

Initially there is no textview. It is added dynamically via code depending on the necessity.

**Comment by hackiftekhar** [comment]
So you are adding UILayoutConstraint from the code. right?

**Comment by PuneetKohli** [comment]
`reloadLayoutIfNeeded` is bouncing the view back to the top.

Yes, but there are potentially a lot of UITextView's ; all of whom are embedded as an item in a Table View. In this case I wouldn't know how to handle individual constraints.

My View Heirarchy looks something like this

UITableView
-> Header View
-> UITextView
-> UIImageView
-> UITextView
-> UIImageView
-> UIImageView
-> UITextView

Basically the user is making notes, and can add an image anywhere in between.

**Comment by hackiftekhar** [comment]
Are you using UITableViewController and the UITableView you shown is UITableViewController.UITableView?

**Comment by PuneetKohli** [comment]
No it is a UIViewController.

There are other views in the scene, so can't use a TableViewController

**Comment by hackiftekhar** [comment]
I think you should share a demo project if you are facing an issue which I'm unable to reproduce. I can check the demo.

**Comment by PuneetKohli** [comment]
I'll try and wrap up a demo project with only said functionality, and get
back to you- thanks!




On Mon, May 23, 2016 at 9:57 PM, Mohd Iftekhar Qurashi <
notifications@github.com> wrote:

> I think you should share a demo project if you are facing an issue which
> I'm unable to reproduce. I can check the demo.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hackiftekhar/IQKeyboardManager/issues/498#issuecomment-221023545>
>



-- 
Thank you,
Puneet Kohli

**Comment by hackiftekhar** [comment]
Any updates for demo project?

**Comment by hackiftekhar** [comment]
@PuneetKohli Do you have any updates for the demo?

**Comment by PuneetKohli** [comment]
Hey sorry,

Haven't been able to put it together - you can close this if you want. I'll
open another issue once I have the demo ready




On Wed, Jun 22, 2016 at 7:50 PM, Mohd Iftekhar Qurashi <
notifications@github.com> wrote:

> @PuneetKohli <https://github.com/PuneetKohli> Do you have any updates for
> the demo?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hackiftekhar/IQKeyboardManager/issues/498#issuecomment-227759094>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIAas9xUaWCdvcWqt8ksZXrLSbckIxcxks5qOUSvgaJpZM4IQUcm>
> .
>



-- 
Thank you,
Puneet Kohli

**Comment by hackiftekhar** [comment]
ok I'm closing this ticket now, but it would be better for the library if you could provide me a demo in near future.
